File: frontend/src/types/episode.ts
export interface PodcastMetadata {
  duration?: string
  publish_time?: string
  episode_title?: string
  podcast_name?: string
  source_url?: string
  extracted_at: string
}

export interface Episode {
  id: string
  title: string
  podcastName: string
  duration: string
  fileSize: number
  downloadDate: string
  showNotes: string
  filePath: string
  coverImagePath?: string
  sourceUrl?: string
  metadata?: PodcastMetadata
}

export interface PaginatedEpisodes {
  episodes: Episode[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

File: backend/internal/models/episode.go
package models

import "time"

// Episode represents a downloaded podcast episode with metadata
type Episode struct {
	ID             string            `json:"id"`
	Title          string            `json:"title"`
	PodcastName    string            `json:"podcastName"`
	Duration       string            `json:"duration"`
	FileSize       int64             `json:"fileSize"`
	DownloadDate   time.Time         `json:"downloadDate"`
	ShowNotes      string            `json:"showNotes"`
	FilePath       string            `json:"filePath"`
	CoverImagePath string            `json:"coverImagePath,omitempty"`
	Metadata       *PodcastMetadata  `json:"metadata,omitempty"`
}

// PaginatedEpisodes represents a paginated response of episodes
type PaginatedEpisodes struct {
	Episodes   []Episode `json:"episodes"`
	Total      int       `json:"total"`
	Page       int       `json:"page"`
	PageSize   int       `json:"pageSize"`
	TotalPages int       `json:"totalPages"`
}
